lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -202,9 +214,13 @@ def _loop_process_output(self, output):
         if not self.seen_output:
             msg = check_output(time_passed, self.output_timeout)
             if msg:
-                self.log.warn(msg)
                 # avoid getting warning multiple times by setting seen_output to True if a warning was produced
                 self.seen_output = True
+                if self.fatal_option:
+                    self.stop_tasks()
+                    raise RunMPIException(TIMEOUT_EXIT_CODE, fatal_msg(msg))
```

Comment:
since you need this in two places, use a function to get both the error message and exit code

```python
error_msg, exit_code = output_timeout_error()
self.log.error(error_msg)
sys.exit(exit_code)
```